VotingApplication/VotingApplication.Web/Scripts/Directives/DatePicker.js

```diff
@@ -0,0 +1,99 @@
+﻿(function () {
+    angular
+        .module('GVA.Creation')
+        .directive('datePicker', datePicker);
+
+    function datePicker() {
+
+        var displayDate = moment();
+        var selectedDate = null;
+        var modelDirty = false;
+
+        function link(scope, element, attrs) {
+            scope.weekdays = moment.weekdaysShort();
+
+            scope.setModelDate = setModelDate;
+            scope.moveMonth = moveMonth;
+
+            activate();
+
+            function setModelDate(date) {
+                modelDirty = true;
+                scope.ngModel = date.toDate();
+            }
+
+            function moveMonth(offset) {
+                displayDate.add(offset, 'months');
+                updateDisplay(displayDate);
+            }
+
+            function updateDisplay(date) {
+                scope.rows = calculateRowDates(date);
+                scope.displayDate = date.format('MMMM YYYY');
+            }
+
+            function activate() {
+                scope.$watch("ngModel", function () {
+                    selectedDate = scope.ngModel ? moment(scope.ngModel) : moment();
+                    displayDate = moment(selectedDate);
+                    updateDisplay(displayDate);
+                    if (scope.update && modelDirty) {
+                        scope.update();
+                        modelDirty = false;
+                    }
+                });
+            }
+        }
+
+        function calculateRowDates(date) {
+
+            rows = [];
+
+            var startDateOfMonth = date.date(1);
+            var daysInMonth = date.daysInMonth();
+            var firstDayOfMonth = startDateOfMonth.day();
+
+            var cellCounter = firstDayOfMonth;
+            var rowCounter = 0;
+            var dateCounter = moment(startDateOfMonth);
+
+            // Fill up empty date cells
+            rows[0] = [];
```

Comment:
Is there a reason this is hardcoded as 0, instead of using rowCounter as the rest of the filling code does?